bst.py

```diff
@@ -0,0 +1,127 @@
+from tree_node import Node
+
+
+class BST:
+    def __init__(self) -> None:
+        self.root = None
+    
+    def add_node(self, element):
+        if not self.root:
+            self.root = Node(element)
+            return True
+        
+        temp = self.root
+        while temp:
+            if element < temp.value:
+                if not temp.left:
+                    temp.left = Node(element)
+                    return True
+                temp = temp.left
+            else:
+                if not temp.right:
+                    temp.right = Node(element)
+                    return True
+                temp = temp.right
+    
+    def pre_order_traversal(self, node):
+        temp = node
+        if temp:
+            print(temp.value)
+            self.pre_order_traversal(temp.left)
+            self.pre_order_traversal(temp.right)
+        
+    def post_order_traversal(self, node):
+        temp = node
+        if temp:
+            self.pre_order_traversal(temp.left)
+            self.pre_order_traversal(temp.right)
+            print(temp.value)
+
+    def in_order_traversal(self, node):
+        temp = node
+        if temp:
+            self.pre_order_traversal(temp.left)
+            print(temp.value)
+            self.pre_order_traversal(temp.right)
+
+    def search_node(self, element):
+        if not self.root:
+            return None
+        temp = self.root
+        while temp:
+            if temp.value == element:
+                return temp
+            elif element < temp.value:
+                temp = temp.left
+            else:
+                temp = temp.right
+
+        return None
+    
+    def delete_node(self, element):
+        node = self.search_node(element)
+        if not node:
+            return None
+        if not node.right and node.left:
+            node.value = None
+        elif not node.right and node.left:
+            pass
+        else:
+            pass
+
+    def max_height(self, source, height=0):
+        temp = source
+        if not temp:
+            return height
+
+        height = max(height, self.max_height(temp.right, height+1), self.max_height(temp.left, height+1))
```

Comment:
What if `source` doesn't exist in the tree?
 `max_height` implies it's the max_height of the tree ie the source is always the root! If you want height from a particular node than that is `relative_height` not `max_height`

Reply:
Got it

Reply:
Either have `root` as value and remove parameter under `max_height` or make it `relative_height`